api/src/types/snapshot.ts

```diff
@@ -13,20 +15,30 @@ export type Snapshot = {
 export type SnapshotField = Field & { meta: Omit<FieldMeta, 'id'> };
 export type SnapshotRelation = Relation & { meta: Omit<RelationMeta, 'id'> };
 
+export type SnapshotWithHash = Omit<Snapshot, 'collections' | 'fields' | 'relations'> & {
```

Comment:
Do we need the omit here if it's overridden by the second object? I thought the `&` worked as an "assign" instead of a "merge"

Reply:
This is definitely not my strong suit, but it seems like removing the `Omit` makes typescript unhappy to build it. Any idea why? 🤔 

I believe I might've actually resorted to `Omit` specifically to resolve this at the time, but unfortunately now I honestly don't remember _why_, really sorry for that 😅

Reply:
Oh yeah that makes it much clearer now. I think that was indeed the original reasoning 👍